2.x Improve documentation for terms

gchtr opened the pull request:
**Ticket**: #1617

## Issue

There were still some todos for terms, and also some issues.

## Solution

- Add changes for terms to the Upgrade Guide.
- Fix some issues that we forgot or changed in the meantime.

## Impact

None.

## Usage Changes

None.

## Considerations

1. In the documentation for [Querying Terms](https://timber.github.io/docs/v2/guides/terms/#querying-terms), I once wrote that there will be `TermCollections` that work similar to `PostCollections`. I guess I’ll have to remove that part, because we don’t have term collections yet?
2. In the same section, I wrote "If you don’t pass in any argument, Timber will use the global query.". Currently, that’s not what we do. Instead, we get terms from all all queryable taxonomies. So, we could probably do something similar to what we already do in https://github.com/timber/timber/blob/41f0d6f61c4121095c6dc810050fc93f85245e68/lib/Timber.php#L350-L354, or we could change the documentation.
3. And I’m thinking: should we set up a default term object in [`Timber::context()`](https://github.com/timber/timber/blob/2.x/lib/Timber.php#L910) as well, if a term is returned by `get_queried_object()`?

## Testing

Only docs, no tests.

jarednova commented:
Sorry, I realized I missed your Qs at the end ...

***In the documentation for Querying Terms, I once wrote that there will be TermCollections that work similar to PostCollections. I guess I’ll have to remove that part, because we don’t have term collections yet?**
Yeah, I think we add nothing right now unless it's absolutely necessary. Full polymorphism should be pursued, but not create a blocker here.

**In the same section, I wrote "If you don’t pass in any argument, Timber will use the global query.". Currently, that’s not what we do. Instead, we get terms from all all queryable taxonomies ...**
Agree that it should function as you wrote it in the docs. That's def. what I'd expect as a user.

**And I’m thinking: should we set up a default term object in Timber::context() as well, if a term is returned by get_queried_object()?**
Yep, similar to above — that's def. what I would expect as a user

gchtr commented on docs/v2/guides/terms.md:
```diff
@@ -90,16 +92,14 @@ You can use this function in a similar way to how you use [`WP_Term_Query`](http
+Also check out the documentation for [`Timber::get_terms()`](https://timber.github.io/docs/v2/reference/timber/#get-terms).
 
-What you get as a return value is not a pure array of posts, but a `Timber\TermCollection` object, which is an `ArrayObject` that is very similar to an array as you know it. To loop over the terms collection in PHP, you first need to convert it to an array with `$terms->get_terms()`.
+What you get as a return value is not a pure array of term, but a `Timber\TermCollection` object, which is an `ArrayObject` that is very similar to an array as you know it. To loop over the terms collection in PHP, you first need to convert it to an array with `$terms->get_terms()`.
```
This whole paragraph is not valid anymore and I replaced it with what’s actually being done, reflecting consideration 1).

> 1. In the documentation for [Querying Terms](https://timber.github.io/docs/v2/guides/terms/#querying-terms), I once wrote that there will be `TermCollections` that work similar to `PostCollections`. I guess I’ll have to remove that part, because we don’t have term collections yet?

gchtr commented:
> ***In the documentation for Querying Terms, I once wrote that there will be TermCollections that work similar to PostCollections. I guess I’ll have to remove that part, because we don’t have term collections yet?**
> Yeah, I think we add nothing right now unless it's absolutely necessary. Full polymorphism should be pursued, but not create a blocker here.

I updated the docs accordingly.

> **In the same section, I wrote "If you don’t pass in any argument, Timber will use the global query.". Currently, that’s not what we do. Instead, we get terms from all all queryable taxonomies ...**
> Agree that it should function as you wrote it in the docs. That's def. what I'd expect as a user.

I figured that there’s (obviously 😅) no concept of a global query for terms in WordPress. The global query always gets posts. You can have templates that filter posts by singular terms. Then you’ll only ever have one term as the queried object (matching `is_category()`, `is_tax()`, etc.).

So, having `get_taxonomies()` as a default for `Timber::get_terms()` makes sense. I added the hint about `get_taxonomies()`.
